File: physio-portal-vue3/src/types/pet.ts
export interface MedicalHistory {
  medicalHistoryId: number
  diagnosis: string
  injuryOrCondition: string | null
  surgeryDate: string | null
  clinicianNotes: string | null
}

export interface Pet {
  petId: number
  ownerId: number
  ownerName: string
  petName: string
  species: string
  breed: string | null
  birthDate: string | null
  weightKg: number | null
  medicalHistories: MedicalHistory[]
}

export interface CreateMedicalHistory {
  diagnosis: string
  injuryOrCondition?: string
  surgeryDate?: string
  clinicianNotes?: string
}

export interface CreateOwner {
  email: string
  firstName: string
  lastName: string
  phoneNumber?: string
  temporaryPassword: string
}

export interface CreatePetRequest {
  ownerId?: number
  petName: string
  species: string
  breed?: string
  birthDate?: string
  weightKg?: number
  initialMedicalHistory?: CreateMedicalHistory
  newOwner?: CreateOwner
}

export interface UpdatePetRequest {
  petName: string
  species: string
  breed?: string
  birthDate?: string
  weightKg?: number
}

export const PET_SPECIES = ['Canine', 'Feline', 'Equine', 'Avian', 'Other'] as const

File: owner-app-flutter/lib/features/pets/models/pet.dart
class PetMedicalHistory {
  const PetMedicalHistory({
    required this.medicalHistoryId,
    required this.diagnosis,
    this.injuryOrCondition,
    this.surgeryDate,
    this.clinicianNotes,
  });

  final int medicalHistoryId;
  final String diagnosis;
  final String? injuryOrCondition;
  final String? surgeryDate;
  final String? clinicianNotes;

  factory PetMedicalHistory.fromJson(Map<String, dynamic> json) {
    return PetMedicalHistory(
      medicalHistoryId: json['medicalHistoryId'] as int,
      diagnosis: json['diagnosis'] as String,
      injuryOrCondition: json['injuryOrCondition'] as String?,
      surgeryDate: json['surgeryDate'] as String?,
      clinicianNotes: json['clinicianNotes'] as String?,
    );
  }
}

class Pet {
  const Pet({
    required this.petId,
    required this.ownerId,
    required this.ownerName,
    required this.petName,
    required this.species,
    this.breed,
    this.birthDate,
    this.weightKg,
    required this.medicalHistories,
  });

  final int petId;
  final int ownerId;
  final String ownerName;
  final String petName;
  final String species;
  final String? breed;
  final String? birthDate;
  final double? weightKg;
  final List<PetMedicalHistory> medicalHistories;

  factory Pet.fromJson(Map<String, dynamic> json) {
    return Pet(
      petId: json['petId'] as int,
      ownerId: json['ownerId'] as int,
      ownerName: json['ownerName'] as String,
      petName: json['petName'] as String,
      species: json['species'] as String,
      breed: json['breed'] as String?,
      birthDate: json['birthDate'] as String?,
      weightKg: (json['weightKg'] as num?)?.toDouble(),
      medicalHistories: (json['medicalHistories'] as List<dynamic>? ?? [])
          .map((item) => PetMedicalHistory.fromJson(item as Map<String, dynamic>))
          .toList(),
    );
  }
}

const petSpecies = ['Canine', 'Feline', 'Equine', 'Avian', 'Other'];
